# HG changeset patch
# User andrew
# Date 1398287189 -3600
#      Wed Apr 23 22:06:29 2014 +0100
# Node ID 6cb711fa264d25f3592471a81a5b865c505a9fd7
# Parent  a8b6c290873cc30d96137bd29724638600317f92
8041658: Use of -fdevirtualize on macroAssembler_x86.o (via -O2) with gcc 4.9.0 creates broken VM
Summary: -fno-devirtualize should be passed when compiling this file.

diff -Nru openjdk.orig/hotspot/make/linux/makefiles/gcc.make openjdk/hotspot/make/linux/makefiles/gcc.make
--- openjdk.orig/hotspot/make/linux/makefiles/gcc.make	2017-04-20 14:31:49.000000000 +0100
+++ openjdk/hotspot/make/linux/makefiles/gcc.make	2017-05-10 14:52:39.297377302 +0100
@@ -259,6 +259,9 @@
   ifeq ($(shell expr $(CC_VER_MAJOR) = 4 \& $(CC_VER_MINOR) = 3), 1)
     OPT_CFLAGS/mulnode.o += $(OPT_CFLAGS/NOOPT)
   endif
+  ifeq ($(shell expr $(CC_VER_MAJOR) = 4 \& $(CC_VER_MINOR) = 9), 1)
+    OPT_CFLAGS/macroAssembler_x86.o += -fno-devirtualize
+  endif
 endif
 
 # Need extra inlining to collapse all the templated closures into the hot loop
